modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/Sequence.scala

```diff
@@ -116,6 +118,52 @@ object Sequence {
           )
       }
 
+    def zipper(seq: Sequence[Action \/ Result]): Option[Zipper] =
+      separate(seq).flatMap {
+        case (pending, done)   => pending match {
+          case Nil             => None
+          case s :: ss =>
+            Step.Zipper.currentify(s).map(
+              Zipper(seq.id, seq.metadata, ss, _, done)
+            )
+        }
+      }
+
+    // We would use MonadPlus' `separate` if we wanted to separate Actions or
+    // Results, but here we want only Steps.
+    private def separate(seq: Sequence[Action \/ Result]): Option[(List[Step[Action]], List[Step[Result]])] = {
+
+      seq.steps.foldLeftM[Option, (List[Step[Action]], List[Step[Result]])]((Nil, Nil))(
+        (acc, step) =>
+        if (Step.status(step) === StepState.Pending)
+          Some(
+            acc.leftMap(
+              _ :+ step.map(
+                _.fold(
+                  identity,
+                  // It should never happen
+                  _ => Task(Result.Error("Inconsistent status"))
```

Comment:
An `Action` is required there by the types, but that code branch should never be reached because I already filtered for all the lefts.

I acknowledge this code sucks, I tried to just get it to work quickly.